eval/adjust.rkt

```diff
@@ -65,16 +68,30 @@
       (or (not useful?)
           (and (<= prec-new prec-old)
                (andmap (lambda (x) (or (< x varc) (vector-ref vrepeats (- x varc)))) (cdr instr)))))
+    (set! any-false? (or any-false? (not repeat)))
     (vector-set! vrepeats n repeat))
 
   ; Step 4. Copying new precisions into vprecs
-  (vector-copy! vprecs 0 vprecs-new))
+  (vector-copy! vprecs 0 vprecs-new)
+
+  ; Step 5. If precisions have not changed but the point didn't converge. A problem exists - add slack to every op
+  (unless any-false?
+    (set-rival-machine-bumps! machine (add1 bumps))
+    (define slack (get-slack))
+    (for ([prec (in-vector vprecs)]
+          [n (in-range (vector-length vprecs))])
+      (define prec* (min (*rival-max-precision*) (+ prec slack)))
+      (when (equal? prec* (*rival-max-precision*))
+        (*last-iteration* #t))
+      (vector-set! vprecs n prec*))
+    (vector-fill! vrepeats #f)))
```

Comment:
Step 5 deals with the following case:
An overflow has been observed but the correct can not be established.
A type-of-benchmark is `(... (log (exp x)))`.
The final result may be close to infinity but does not round to infinity yet.
The result may look like `[1e200, inf]`, where the high endpoint is _fixed_.
After few iterations all the amplifications converge but the correct result has not been obtained yet and
the precisions stay the same.

The proper way to handle this case is:
Increase precisions with `slack` for the operations that has over/underflowed.
Increase it, until max precision has been reached.

Once the max precision has been reached - it means that the next following iteration 
precisions just can not go any higher, repeats optimization marks all the operations as useless to reevaluate.
At this step we can exit knowing that nothing better can be done